src/HTSPDemuxer.cpp

```diff
@@ -210,12 +210,16 @@ void CHTSPDemuxer::Weight ( enum eSubscriptionWeight weight )
 PVR_ERROR CHTSPDemuxer::CurrentStreams ( PVR_STREAM_PROPERTIES *props )
 {
   CLockObject lock(m_mutex);
-  for (int i = 0; i < m_streams.iStreamCount; i++)
+
+  /* Truncate the list of streams to prevent overflows */
+  m_streams.resize(PVR_STREAM_MAX_STREAMS);
+
+  for (size_t i = 0; i < m_streams.size() && i < PVR_STREAM_MAX_STREAMS; i++)
```

Comment:
The <code>resize</code> some lines above is imo not needed, as you already prevent overflows here.